docs/docs/contributing/developer-guide.md

```diff
@@ -33,8 +33,34 @@ As of now, we only support integration tests who interact directly with Aiven. T
 an [Aiven account](https://console.aiven.io/signup?utm_source=github&utm_medium=organic&utm_campaign=k8s-operator&utm_content=signup)
 and an [Aiven authentication code](https://help.aiven.io/en/articles/2059201-authentication-tokens).
 
+### Prerequisites
+
+Please have installed first:
+
+- docker/podman
+- [kind](https://kind.sigs.k8s.io/)
+- [helm](https://helm.sh/)
+- [Aiven CLI](https://docs.aiven.io/docs/tools/cli) (make sure you have logged in)
+- [jq](https://stedolan.github.io/jq/)
+- [kcat](https://github.com/edenhill/kcat)
+- base64 (note: MACOS version doesn't support `-w0` flag, some tests may not work properly)
+
+Create a cluster:
+
+```shell
+ kind create cluster --image kindest/node:v1.24.0 --wait 5m
+```
+
+Setup everything:
+
+```shell
+AIVEN_TOKEN="<your-token>" AIVEN_PROJECT_NAME="<your-project-name>" make e2e-setup-kind
```

Comment:
Now I get this 
```
make e2e-setup-kind
docker build -t operator:62e97ab1e5134ef309ab8cb2acd7b7c6de8c1657 .
[+] Building 1.0s (17/17) FINISHED                                                                                  
 => [internal] load .dockerignore                                                                              0.1s
 => => transferring context: 230B                                                                              0.0s
 => [internal] load build definition from Dockerfile                                                           0.2s
 => => transferring dockerfile: 938B                                                                           0.0s
 => [internal] load metadata for gcr.io/distroless/static:nonroot                                              0.7s
 => [internal] load metadata for docker.io/library/golang:1.18                                                 0.7s
 => [stage-1 1/3] FROM gcr.io/distroless/static:nonroot@sha256:149531e38c7e4554d4a6725d7d70593ef9f98813588094  0.0s
 => [internal] load build context                                                                              0.1s
 => => transferring context: 16.85kB                                                                           0.0s
 => [builder 1/9] FROM docker.io/library/golang:1.18@sha256:50c889275d26f816b5314fc99f55425fa76b18fcaf16af255  0.0s
 => CACHED [builder 2/9] WORKDIR /workspace                                                                    0.0s
 => CACHED [builder 3/9] COPY go.mod go.mod                                                                    0.0s
 => CACHED [builder 4/9] COPY go.sum go.sum                                                                    0.0s
 => CACHED [builder 5/9] RUN go mod download                                                                   0.0s
 => CACHED [builder 6/9] COPY main.go main.go                                                                  0.0s
 => CACHED [builder 7/9] COPY api/ api/                                                                        0.0s
 => CACHED [builder 8/9] COPY controllers/ controllers/                                                        0.0s
 => CACHED [builder 9/9] RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go              0.0s
 => CACHED [stage-1 2/3] COPY --from=builder /workspace/manager .                                              0.0s
 => exporting to image                                                                                         0.0s
 => => exporting layers                                                                                        0.0s
 => => writing image sha256:554910de67e02a9124b4c47378e637b64051cad02fe9a2ce9b14d0fad2db1313                   0.0s
 => => naming to docker.io/library/operator:62e97ab1e5134ef309ab8cb2acd7b7c6de8c1657                           0.0s
kind load docker-image localhost/operator:62e97ab1e5134ef309ab8cb2acd7b7c6de8c1657
ERROR: image: "localhost/operator:62e97ab1e5134ef309ab8cb2acd7b7c6de8c1657" not present locally
make: *** [Makefile:312: e2e-setup-kind] Error 1
```

Reply:
Where do I remove this? Can you please update the PR?

Reply:
If I completely remove it from the command I get this: 
```
make e2e-setup-kind
docker build -t operator:62e97ab1e5134ef309ab8cb2acd7b7c6de8c1657 .
[+] Building 1.0s (17/17) FINISHED                                                                                  
 => [internal] load .dockerignore                                                                              0.1s
 => => transferring context: 230B                                                                              0.0s
 => [internal] load build definition from Dockerfile                                                           0.2s
 => => transferring dockerfile: 938B                                                                           0.0s
 => [internal] load metadata for gcr.io/distroless/static:nonroot                                              0.7s
 => [internal] load metadata for docker.io/library/golang:1.18                                                 0.7s
 => [stage-1 1/3] FROM gcr.io/distroless/static:nonroot@sha256:149531e38c7e4554d4a6725d7d70593ef9f98813588094  0.0s
 => [internal] load build context                                                                              0.1s
 => => transferring context: 16.85kB                                                                           0.0s
 => [builder 1/9] FROM docker.io/library/golang:1.18@sha256:50c889275d26f816b5314fc99f55425fa76b18fcaf16af255  0.0s
 => CACHED [builder 2/9] WORKDIR /workspace                                                                    0.0s
 => CACHED [builder 3/9] COPY go.mod go.mod                                                                    0.0s
 => CACHED [builder 4/9] COPY go.sum go.sum                                                                    0.0s
 => CACHED [builder 5/9] RUN go mod download                                                                   0.0s
 => CACHED [builder 6/9] COPY main.go main.go                                                                  0.0s
 => CACHED [builder 7/9] COPY api/ api/                                                                        0.0s
 => CACHED [builder 8/9] COPY controllers/ controllers/                                                        0.0s
 => CACHED [builder 9/9] RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go              0.0s
 => CACHED [stage-1 2/3] COPY --from=builder /workspace/manager .                                              0.0s
 => exporting to image                                                                                         0.0s
 => => exporting layers                                                                                        0.0s
 => => writing image sha256:554910de67e02a9124b4c47378e637b64051cad02fe9a2ce9b14d0fad2db1313                   0.0s
 => => naming to docker.io/library/operator:62e97ab1e5134ef309ab8cb2acd7b7c6de8c1657                           0.0s
kind load docker-image localhost/operator:62e97ab1e5134ef309ab8cb2acd7b7c6de8c1657
ERROR: image: "localhost/operator:62e97ab1e5134ef309ab8cb2acd7b7c6de8c1657" not present locally
make: *** [Makefile:312: e2e-setup-kind] Error 1
➜  aiven-operator git:(byashimov-install-in-kind) make e2e-setup-kind
docker build -t operator:62e97ab1e5134ef309ab8cb2acd7b7c6de8c1657 .
[+] Building 1.5s (17/17) FINISHED                                                                                  
 => [internal] load .dockerignore                                                                              0.1s
 => => transferring context: 230B                                                                              0.0s
 => [internal] load build definition from Dockerfile                                                           0.2s
 => => transferring dockerfile: 938B                                                                           0.0s
 => [internal] load metadata for docker.io/library/golang:1.18                                                 1.2s
 => [internal] load metadata for gcr.io/distroless/static:nonroot                                              0.6s
 => [builder 1/9] FROM docker.io/library/golang:1.18@sha256:50c889275d26f816b5314fc99f55425fa76b18fcaf16af255  0.0s
 => [internal] load build context                                                                              0.1s
 => => transferring context: 16.85kB                                                                           0.0s
 => [stage-1 1/3] FROM gcr.io/distroless/static:nonroot@sha256:149531e38c7e4554d4a6725d7d70593ef9f98813588094  0.0s
 => CACHED [builder 2/9] WORKDIR /workspace                                                                    0.0s
 => CACHED [builder 3/9] COPY go.mod go.mod                                                                    0.0s
 => CACHED [builder 4/9] COPY go.sum go.sum                                                                    0.0s
 => CACHED [builder 5/9] RUN go mod download                                                                   0.0s
 => CACHED [builder 6/9] COPY main.go main.go                                                                  0.0s
 => CACHED [builder 7/9] COPY api/ api/                                                                        0.0s
 => CACHED [builder 8/9] COPY controllers/ controllers/                                                        0.0s
 => CACHED [builder 9/9] RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go              0.0s
 => CACHED [stage-1 2/3] COPY --from=builder /workspace/manager .                                              0.0s
 => exporting to image                                                                                         0.0s
 => => exporting layers                                                                                        0.0s
 => => writing image sha256:554910de67e02a9124b4c47378e637b64051cad02fe9a2ce9b14d0fad2db1313                   0.0s
 => => naming to docker.io/library/operator:62e97ab1e5134ef309ab8cb2acd7b7c6de8c1657                           0.0s
kind load docker-image operator:62e97ab1e5134ef309ab8cb2acd7b7c6de8c1657
Image: "operator:62e97ab1e5134ef309ab8cb2acd7b7c6de8c1657" with ID "sha256:554910de67e02a9124b4c47378e637b64051cad02fe9a2ce9b14d0fad2db1313" not yet present on node "kind-control-plane", loading...
helm uninstall aiven-operator || true
release "aiven-operator" uninstalled
helm uninstall aiven-operator-crds || true
release "aiven-operator-crds" uninstalled
helm uninstall cert-manager || true
Error: uninstall: Release not loaded: cert-manager: release: not found
kubectl delete secret aiven-token || true
secret "aiven-token" deleted
kubectl delete -f https://github.com/cert-manager/cert-manager/releases/download/v1.9.0/cert-manager.crds.yaml || true
Error from server (NotFound): error when deleting "https://github.com/cert-manager/cert-manager/releases/download/v1.9.0/cert-manager.crds.yaml": customresourcedefinitions.apiextensions.k8s.io "certificaterequests.cert-manager.io" not found
Error from server (NotFound): error when deleting "https://github.com/cert-manager/cert-manager/releases/download/v1.9.0/cert-manager.crds.yaml": customresourcedefinitions.apiextensions.k8s.io "certificates.cert-manager.io" not found
Error from server (NotFound): error when deleting "https://github.com/cert-manager/cert-manager/releases/download/v1.9.0/cert-manager.crds.yaml": customresourcedefinitions.apiextensions.k8s.io "challenges.acme.cert-manager.io" not found
Error from server (NotFound): error when deleting "https://github.com/cert-manager/cert-manager/releases/download/v1.9.0/cert-manager.crds.yaml": customresourcedefinitions.apiextensions.k8s.io "clusterissuers.cert-manager.io" not found
Error from server (NotFound): error when deleting "https://github.com/cert-manager/cert-manager/releases/download/v1.9.0/cert-manager.crds.yaml": customresourcedefinitions.apiextensions.k8s.io "issuers.cert-manager.io" not found
Error from server (NotFound): error when deleting "https://github.com/cert-manager/cert-manager/releases/download/v1.9.0/cert-manager.crds.yaml": customresourcedefinitions.apiextensions.k8s.io "orders.acme.cert-manager.io" not found
kubectl delete namespace cert-manager || true
Error from server (NotFound): namespaces "cert-manager" not found
kubectl create secret generic aiven-token --from-literal=token=fFXW/5LzFkIH9xAUIcQrzKg0ni4PqmQwPiHJzAshRoeZ64ip7XwAJ1X8aEah4cQ+V5JypD8i3UY3huNv3IFEuo+S9ktvQ8c2mf8H19NEl3F1tqzq1fwW6pIdwUhm1M0x0hSXg61dOW9go9Zjun0UNFdEMrcQOEh1cVSdtS6+D3tOC4S5BLuZEQCSonczuhUUcAgoKY6fdo+bQwuvTyGW1af0uNkLED6irNft8BJd/jCi0IqZFHFfZS74coKvV6ogCegYzfLwHdf0MbSwtfgpaxKb6PIqC2NUMw/OihyNTXfViDfwQi3RySjeqRNvNDIYuL1xn3VsKiJZgyq0KvMrv0PSjCmsQ6yiB4Yd2UkfZGFeFFaQ || true
secret/aiven-token created
kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.9.0/cert-manager.crds.yaml
customresourcedefinition.apiextensions.k8s.io/certificaterequests.cert-manager.io created
customresourcedefinition.apiextensions.k8s.io/certificates.cert-manager.io created
customresourcedefinition.apiextensions.k8s.io/challenges.acme.cert-manager.io created
customresourcedefinition.apiextensions.k8s.io/clusterissuers.cert-manager.io created
customresourcedefinition.apiextensions.k8s.io/issuers.cert-manager.io created
customresourcedefinition.apiextensions.k8s.io/orders.acme.cert-manager.io created
helm install cert-manager jetstack/cert-manager \
        --namespace cert-manager \
        --create-namespace \
        --version v1.9.0
Error: INSTALLATION FAILED: repo jetstack not found
make: *** [Makefile:321: e2e-setup-kind] Error 1
```

Reply:
Updated, plz try again.

Reply:
ok, works now